file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemProvider.kt

```diff
@@ -218,13 +220,31 @@ internal class AttributeCachingFileSystemProvider : FileSystemProvider() {
         return actualPath.fileSystem.provider().checkAccess(actualPath, *modes)
     }
 
+    /**
+     * Get the [FileAttributeView] specified by the `Class` [type] from the incoming [path].
+     *
+     * If the requested [type] is [AclFileAttributeView] and we have an incoming [AttributeCachingPath], we get the
+     * cached view.
+     *
+     * @param path The [Path] to read the [FileAttributeView] from.
+     * @param type The `Class` of the [FileAttributeView] to be read. `Class` types include [BasicFileAttributeView],
+     * [DosFileAttributeView], [PosixFileAttributeView], [AclFileAttributeView], or [UserDefinedFileAttributeView].
+     * @param options The [LinkOption]s indicating how symbolic links are handled.
+     * @return The [FileAttributeView] for the given [path] or `null` if the requested view cannot be found or is not
+     * supported.
+     */
     override fun <V : FileAttributeView?> getFileAttributeView(
         path: Path,
         type: Class<V>?,
         vararg options: LinkOption?
     ): V {
         val actualPath = path.getUnderlyingPath()
-        return actualPath.fileSystem.provider().getFileAttributeView(actualPath, type, *options)
+        return if (path is AttributeCachingPath && type is AclFileAttributeView) {
+            @Suppress("UNCHECKED_CAST")
+            path.aclFileAttributeView as V
+        } else {
+            actualPath.fileSystem.provider().getFileAttributeView(actualPath, type, *options)
```

Comment:
yeah umm then this becomes very very complicated as these objects are completely different than the other file attributes.

Reply:
Just as an update, while I attached a screenshot of a JIMFS class, that as I suspected it might, the Windows implementation unfortunately shares this problem.

<img width="761" alt="image" src="https://github.com/pkware/AttributeCachingFileSystem/assets/77342611/5caf8ed2-0e7e-45c9-b42e-4420f766f377">

Reply:
yeah i am just going with caching the individual acl attributes, not going to make a separate class to hold them both though- theres no need

Reply:
> yeah i am just going with caching the individual acl attributes, not going to make a separate class to hold them both though- theres no need

Agreed, I was just using that point to illustrate that it is trivial to accomplish. Absolutely fine to do it the way you suggested.